src/compile.rs

```diff
@@ -25,6 +26,7 @@ struct Patch {
 
 /// A compiler translates a regular expression AST to a sequence of
 /// instructions. The sequence of instructions represents an NFA.
+#[derive(Debug)]
```

Comment:
Is this `Debug` impl necessary?

Reply:
`Compiler` is public via the internal module which "exists to support suspicious activity".

I'll remove the debug impl and allow the lint.

Reply:
Haha right. Yeah, it's technically public, but is `doc(hidden)` and not a stable part of the API. I'm hoping to remove that quirk in the move to `regex-automata`.

Thanks.